MariaDB/server
# MDEV-27033: Remove version suffix from Debian packages

- [x] *The Jira issue number for this PR is: MDEV-27033*

## Description
Remove version suffix from Debian packages (for example mariadb-server-10.8)
because installing suffixed package removes older version
of package even if it's suffixed (for example mariadb-server-10.7)

This make also Debian package management easier in future MariaDB
version iterations because there is no need for stacking
Conlicts/Breaks/Replaces-parameters in every new major release

## How can this PR be tested?
Launch Debian or Ubuntu instance. Run `debian/autobake-deb.sh` script to build deb packages and install created packages over the current Debian or Ubuntu packages

This can be autotested but should be make sure in Buildbot instances that it uses correct version system.

## Basing the PR against the correct MariaDB version

## Backward compatibility
All the older control file Breaks, Conflict and Replaces are in place so it will work as suffixed version and remove older versions. There also comparability Provides with suffix. Metapackage for server and client is removed as it's not needed anymore
